### PR TITLE
use compose-go version fixing extra_hosts unicity issue

### LINKED ISSUE #12155: [BUG] <extra_hosts array items[0,1] must be unique>
### Description

Docker compose file is seen as invalid.

code is:

```
  foo:
    <<: *base
    container_name: foo
    build:
      context: ../foo
      <<: *docker-build-args
    command: [ 'nodemon', 'entrypoints/bugs' ]
    dns:
      - '8.8.8.8'
    dns_search: internal.namespace
    extra_hosts:
      - 'host.docker.internal:host-gateway'
```

### Steps To Reproduce

```
$> docker compose version
Docker Compose version v2.29.6
```

error message:

validating /home/$USER/Projects/docker-compose.yml: services.foo.extra_hosts array items[0,1] must be unique

### Compose Version

```Text
v2.29.6
```

### Docker Environment

```Text
$> docker version
Client: Docker Engine - Community
 Version:           27.3.0
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        e85edf8
 Built:             Thu Sep 19 14:25:51 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          27.3.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Thu Sep 19 14:25:51 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.22
  GitCommit:        7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc:
  Version:          1.1.14
  GitCommit:        v1.1.14-0-g2c9f560
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Anything else?

Linked to https://github.com/docker/compose/issues/11371

### PR DESCRIPTION
**Pull request by glours**
**What I did**
Bump `compose-go` version fixing `extra_hosts` unicity issue

**Related issue**
fixes #12155 

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -192,3 +192,5 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 	tags.cncf.io/container-device-interface v0.8.0 // indirect
 )
+
+replace github.com/compose-spec/compose-go/v2 => github.com/glours/compose-go/v2 v2.0.0-20240924124937-38f94a79402c
```
Comment (ndeloof) [inline go.mod:196]: ?

Comment (glours) [reply go.mod:196]: Just to test again Compose CI until we release a new version of compose-go

Comment (glours) [reply go.mod:196]: I marked the PR as draft for now

Comment (ndeloof) [reply go.mod:196]: right, I got confused for this reason as I got notified on this one first :)

**Review by jhrotko — APPROVED** [review]

**Pull request merged**